src/core/events/types/common.py

```diff
@@ -60,28 +66,40 @@ def explain(self):
         return self.__doc__
 
 
+global event_id_count_lock
+event_id_count_lock = threading.Lock()
+event_id_count_lock.acquire()
```

Comment:
You don't need to grab the lock here, as this is only run once when the module is loaded